src/vs/workbench/services/runtimeSession/common/runtimeSession.ts

```diff
@@ -475,6 +475,24 @@ export class RuntimeSessionService extends Disposable implements IRuntimeSession
 		);
 	}
 
+	/**
+	 * Focus a runtime session by setting it as the foreground session.
+	 */
+	focusSession(sessionId: string): void {
+		const session = this.getSession(sessionId);
+		if (!session) {
+			this._logService.error(`Could not find session with id {sessionId}.`);
+			return;
+		}
+
+		if (session.metadata.sessionMode === LanguageRuntimeSessionMode.Console) {
+			this.foregroundSession = session;
+		} else {
+			// TODO: we could potentially focus the notebook editor in this case.
+			this._logService.error(`Cannot focus a notebook session.`);
```

Comment:
Should this error be thrown rather than logged? (so the caller knows that the focus failed)

Reply:
Done in https://github.com/posit-dev/positron/pull/7024/commits/0f10376206e68e4d5d39c7fb7a7bc3298a085ece